ext/openssl/ossl_ssl.c

```diff
@@ -1406,6 +1406,28 @@ ossl_ssl_s_alloc(VALUE klass)
     return TypedData_Wrap_Struct(klass, &ossl_ssl_type, NULL);
 }
 
+static VALUE
+peer_ip_address(VALUE self)
+{
+    VALUE remote_address = rb_funcall(rb_attr_get(self, id_i_io), rb_intern("remote_address"), 0);
+    VALUE sockaddr = rb_funcall(remote_address, rb_intern("inspect_sockaddr"), 0);
+
+    if (!RTEST(sockaddr)) {
+        sockaddr = rb_str_new_cstr("(null)");
+    }
```

Comment:
This should go to `peeraddr_ip_str()` (outside `rb_ensure`).

Reply:
I've moved it to the rescue block.